core/src/main/java/com/sequenceiq/cloudbreak/controller/validation/template/TemplateValidator.java

```diff
@@ -187,16 +197,39 @@ private boolean isComputeInstanceGroup(InstanceGroup instanceGroup) {
         return GROUP_NAME_COMPUTE.equalsIgnoreCase(instanceGroup.getGroupName());
     }
 
+    private boolean isCoordinatorInstanceGroup(Stack stack) {
```

Comment:
Both `isCoordinatorInstanceGroup` and `isExecutorInstanceGroup` checks for the same thing. Please merge both these methods.